### PR TITLE
Nexus 20713 cleanup policy

### PR DESCRIPTION
**Pull request by aornatovskyy**
Ticket: https://issues.sonatype.org/browse/NEXUS-20713

### CODE REVIEW

File: nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/cleanup/CleanupTaskRIT.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        versionsOfComponentsToKeep);
+  }
+
+  @Override
+  protected boolean componentMatchesByVersion(final Component component, final String version) {
+    return version
+        .equals(format("%s_%s%s", component.name().toLowerCase(), component.version().toLowerCase(), TARGZ_EXT));
+  }
+
+  private int deployArtifacts(final String... names) {
```
Comment (sergeyonofreychuck) [inline nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/cleanup/CleanupTaskRIT.java]: return value never used

**Comment by sergeyonofreychuck** [comment]
+0  I dont see any issues with the code, but I`m not familiar with cleanup policies tests

File: nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/cleanup/RCleanupPolicyConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import org.sonatype.nexus.repository.r.internal.RFormat;
+
+import com.google.common.collect.ImmutableMap;
+
+import static org.sonatype.nexus.repository.search.DefaultComponentMetadataProducer.IS_PRERELEASE_KEY;
+import static org.sonatype.nexus.repository.search.DefaultComponentMetadataProducer.LAST_BLOB_UPDATED_KEY;
+import static org.sonatype.nexus.repository.search.DefaultComponentMetadataProducer.LAST_DOWNLOADED_KEY;
+import static org.sonatype.nexus.repository.search.DefaultComponentMetadataProducer.REGEX_KEY;
+
+@Named(RFormat.NAME)
```
Comment (doddi) [inline nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/cleanup/RCleanupPolicyConfiguration.java]: Trivial: Missing `@ since`

Comment (aornatovskyy) [reply nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/cleanup/RCleanupPolicyConfiguration.java]: d94cfeee9e6b572acc1ab2ececfbc3dd02c5ed93

Comment (aornatovskyy) [reply nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/cleanup/RCleanupPolicyConfiguration.java]: d94cfeee9e6b572acc1ab2ececfbc3dd02c5ed93

**Pull request merged**